File: sol/1695/1695.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        prevIdx = dict()
        l = -1  # (l, r]
        total = 0
        for r, x in enumerate(nums):
            if (idx := prevIdx.get(x, -1)) > l:
                total -= sum(nums[l + 1:idx + 1])
                l = idx
            total += x
            prevIdx[x] = r
            res = max(res, total)
        return res

File: sol/1695/1695.cpp
struct Solution {
    int maximumUniqueSubarray(vector<int>& nums) {
        int n = nums.size(), res = 0, l = -1, total = 0;
        unordered_map<int, int> prevIdx;
        for (int i = 0; i < n; ++i){
            int x = nums[i];
            if (prevIdx.contains(x) && prevIdx[x] > l){
                for (int j = l + 1; j <= prevIdx[x]; ++j){
                    total -= nums[j];
                }
                l = prevIdx[x];
            }
            total += x;
            prevIdx[x] = i;
            res = max(res, total);
        }
        return res;
    }
};
